packages/server/src/modules/retail-location/retail-location.resolver.ts

```diff
@@ -133,4 +137,242 @@ export class RetailLocationResolver {
+          book: {
+            retailLocationId,
+          },
+          problems: {
+            some: {
+              resolvedAt: null,
+            },
+          },
+          returnedAt: {
+            not: null,
+          },
+        },
+      }),
+    });
+
+    const getTotalPresentBooks = async () => ({
+      totalPresentBooks: await this.prisma.bookCopy.count({
+        where: {
+          book: {
+            retailLocationId,
+          },
+          returnedAt: null,
+          AND: [
+            {
+              OR: this.noProblemsFilter,
+            },
+            {
+              OR: this.notSoldOrRefunded,
+            },
+          ],
+        },
+      }),
+    });
+
+    const getTotalSoldBooks = async () => ({
+      totalSoldBooks: await this.prisma.sale.count({
+        where: {
+          refundedAt: null,
+          bookCopy: {
+            book: {
+              retailLocationId,
+            },
+          },
+        },
+      }),
+    });
+
+    const getTotalReservedBooks = async () => ({
+      totalReservedBooks: await this.prisma.reservation.count({
+        where: {
+          book: {
+            retailLocationId,
+          },
+          deletedAt: null,
+        },
+      }),
+    });
+
+    const getTotalRequestedBooks = async () => ({
+      totalRequestedBooks: await this.prisma.bookRequest.count({
+        where: {
+          book: {
+            retailLocationId,
+          },
+          deletedAt: null,
+        },
+      }),
+    });
+
+    const getTotalUsers = async () => ({
+      totalUsers: await this.prisma.user.count({
+        where: {
+          OR: [
+            {
+              requestedBooks: {
+                some: {
+                  book: {
+                    retailLocationId,
+                  },
+                  deletedAt: null,
+                },
+              },
+            },
+            {
+              bookCopies: {
+                some: {
+                  book: {
+                    retailLocationId,
+                  },
+                },
+              },
+            },
+          ],
+        },
+      }),
+    });
+
+    const getActiveSales = async () => ({
+      activeSales: await this.prisma.sale.findMany({
+        where: {
+          refundedAt: null,
+          bookCopy: {
+            book: {
+              retailLocationId,
+            },
+          },
+        },
+        include: {
+          bookCopy: {
+            select: {
+              settledAt: true,
+              book: {
+                select: {
+                  originalPrice: true,
+                },
+              },
+            },
+          },
+        },
+      }),
+    });
+
+    const getTotals = async () => {
+      const [{ activeSales }, { sellRate, buyRate }] = await Promise.all([
+        getActiveSales(),
+        this.retailLocation({
+          id: retailLocationId,
+        }),
+      ]);
+
+      // TODO: Include discount
+      let totalRevenue = 0;
```

Comment:
For the future, these kind of statistics are usually faster to get by using SQL directly
Which means either creating 3 different ORM calls, but which perform operations at SQL level (e.g. using `_sum`), or for complex cases which require ternaries, as this case, just use raw SQL queries

Do not update this, just know that using raw SQL is usually a better choice for statistics, as the performance is usually way worse than other calls

Reply:
Could you send me a link which explains that usage?

Reply:
Honestly, I couldn't find it either, but it's just breifly mentioned into the docs when talking about aggregations

Reply:
Like [this one](https://dgraph.io/docs/graphql/queries/aggregate/)?

Reply:
That's GraphQL, wtf, we're talking about databases here

Here: https://www.prisma.io/docs/orm/prisma-client/queries/aggregation-grouping-summarizing